manageiq-operator/pkg/controller/manageiq/manageiq_controller.go

```diff
@@ -540,6 +577,52 @@ func (r *ReconcileManageIQ) manageCR(cr *miqv1alpha1.ManageIQ) error {
 	return nil
 }
 
+func (r *ReconcileManageIQ) createSecretWatchers(cr *miqv1alpha1.ManageIQ) error {
+	if cr.Spec.HttpdAuthConfig != "" {
+		secret := &corev1.Secret{}
+		secretErr := r.client.Get(context.TODO(), types.NamespacedName{Namespace: cr.Namespace, Name: cr.Spec.HttpdAuthConfig}, secret)
+		if secretErr != nil {
+			return nil
+		}
+		mapFn := handler.ToRequestsFunc(
+			func(a handler.MapObject) []reconcile.Request {
+				mappedName := a.Meta.GetName()
+				if mappedName == cr.Spec.HttpdAuthConfig {
+					mappedName = "httpd-auth-config-secret-" + cr.Name + "-" + mappedName
+				}
+				return []reconcile.Request{
+					{NamespacedName: types.NamespacedName{
+						Name:      mappedName,
+						Namespace: a.Meta.GetNamespace(),
+					}},
+				}
+			})
+
+		predicateFn := predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				return e.MetaNew.GetName() == cr.Spec.HttpdAuthConfig
+			},
+			CreateFunc: func(e event.CreateEvent) bool {
+				return e.Meta.GetName() == cr.Spec.HttpdAuthConfig
+			},
+		}
+
+		err := controller_manageiq.Watch(&source.Kind{Type: &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: cr.Namespace,
+				Name:      cr.Spec.HttpdAuthConfig,
+			},
+		}}, &handler.EnqueueRequestsFromMapFunc{
+			ToRequests: mapFn,
+		}, predicateFn)
+
+		if err != nil {
+			return err
+		}
```

Comment:
What do you think of something like this:
```suggestion
    authConfigSecret := &corev1.Secret{
      ObjectMeta: metav1.ObjectMeta{
        Namespace: cr.Namespace,
        Name:      cr.Spec.HttpdAuthConfig,
      },
    }

    mapFunc := &handler.EnqueueRequestsFromMapFunc{
      ToRequests: mapFn,
    }

    if err := controller_manageiq.Watch(&source.Kind{Type: authConfigSecret}, mapFunc, predicateFn); err != nil {
      return err
    }
```